### PR TITLE
SW-278 Add API endpoints for facility automations

### PR DESCRIPTION
**Pull request by sgrimm**
The device manager will support various automations that can be run at a
facility. Each automation needs automation-specific configuration data. Add a
series of CRUD endpoints to manage the data.

The automations are stored in a new `automations` table. The Flyway script that
creates that table introduces something we should have been using all along:
instead of having a separate SQL file for Postgres and non-Postgres databases
where the only difference is the type of the JSONB column, instead it uses
Flyway's built-in variable substitution feature and defines a variable for
the column type that should be used for JSON data.

### CODE REVIEW

File: src/main/kotlin/com/terraformation/backend/customer/model/AutomationModel.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    val createdTime: Instant,
+    val modifiedTime: Instant,
+    val configuration: Map<String, Any?>?
+) {
+  constructor(
+      row: AutomationsRow,
+      objectMapper: ObjectMapper
+  ) : this(
+      id = row.id ?: throw IllegalArgumentException("id is required"),
+      facilityId = row.facilityId ?: throw IllegalArgumentException("facilityId is required"),
```
Comment (kallen07) [inline src/main/kotlin/com/terraformation/backend/customer/model/AutomationModel.kt:24]: Will this end up throwing a 5xx or a 4xx error if some data is missing when we create an AutomationModel instance inside the AutomationStore?

File: src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt [hunk 2]
```diff
@@ -406,6 +438,20 @@ internal class PermissionTest : DatabaseTest() {
         updateAccession = true,
     )
 
+    permissions.expect(
```
Comment (kallen07) [inline src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt:441]: What do you think about storing the list of IDs as a variable here? And then constructing the list of Accession IDs, Automation IDs, Device IDs, etc from that list?

File: src/main/kotlin/com/terraformation/backend/customer/model/AutomationModel.kt (hunk 1 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/customer/model/AutomationModel.kt:24]: It'll be a 5xx, which seems like what we'd want.

File: src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt (hunk 2 above)
Comment (sgrimm) [reply src/test/kotlin/com/terraformation/backend/customer/model/PermissionTest.kt:441]: Oh, that's a good idea. I'll do that in a separate PR.

**Pull request merged**